lib/portage/dbapi/vartree.py

```diff
@@ -3764,10 +3754,16 @@ def path_to_node(path):
                 lib_graph.add(preserved_node, None)
                 preserved_paths.add(f)
                 preserved_nodes.add(preserved_node)
-                for c in self.vartree.dbapi._linkmap.findConsumers(f):
-                    consumer_node = path_to_node(c)
-                    if not consumer_node.file_exists():
-                        continue
+                consumer_nodes = (
+                    path_to_node(consumer)
+                    for consumer in self.vartree.dbapi._linkmap.findConsumers(f)
+                )
+                consumer_nodes = (
+                    consumer_node
+                    for consumer_node in consumer_nodes
+                    if consumer_node.file_exists()
+                )
```

Comment:
Is this really more readable than before? It also adds two more loops over this data.

Reply:
The readability here might be questionable, but what's going on here is I'm embedding a generator in another one as one final generator. Then the final generator will run itself sequentially for the following for loop. Both expressions are not eagerly evaluated, so no actual looping happens with either of them.